packages/@react-spectrum/menu/src/SafeTriangle.tsx

```diff
@@ -0,0 +1,85 @@
+/*
+ * Copyright 2020 Adobe. All rights reserved.
+ * This file is licensed to you under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License. You may obtain a copy
+ * of the License at http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under
+ * the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+ * OF ANY KIND, either express or implied. See the License for the specific language
+ * governing permissions and limitations under the License.
+ */
+
+import React, {MutableRefObject, ReactElement, useEffect, useState} from 'react';
+
+interface SafeTriangleProps {
+  /** Ref for the submenu. */
+  subMenuRef: MutableRefObject<any>,
+  /** Ref for the submenu's trigger element. */
+  triggerRef: MutableRefObject<HTMLLIElement>
+}
+
+/**
+ * Allows the user to move their pointer to the sub-menu without it closing
+ * due to the mouse leaving the trigger element.
+ */
+export function SafeTriangle(props: SafeTriangleProps): ReactElement {
```

Comment:
Question for the team: Would we want a react-aria hook for this? Or a RAC feature?

Reply:
Good question, I'm in favor of it. Hook if possible, but we don't support the sub-menu/unavailable items in RAC yet, so might be difficult to do in this PR?

Reply:
Yeah, I think we'll just include it in React Spectrum for now for that reason, but it should be easy to break out in the future for RAC.